add guide on how to add a separate website maintained by Sourcegraph (#6017)
sourcegraph/sourcegraph: doc/dev/documentation/separate_website.md

# How to setup a separate website maintained by Sourcegraph

    💡 Use this template to describe the steps engineers should follow to setup a separate website maintained by Sourcegraph. 

# 1. When do we need a separate website?

For projects like [langserver.org](http://langserver.org) or [lsif.dev](http://lsif.dev), we prefer to create separate websites from 
[Sourcegraph.com](http://sourcegraph.com). In these two examples, we provide a list of LSPs or LSIF indexers offered by us or other parties 

> that can be used for Sourcegraph or other static analysis applications. Since the usage isn't strictly tide to Sourcegraph and we are not 

Reviewer: ```suggestion
that can be used for Sourcegraph or other static analysis applications. Since the usage isn't strictly tied to Sourcegraph and we are not 
```

the sole providers, we publish this information on an independent website.

> # 2. Setup domain and create content

Reviewer: ```suggestion
# 2. Set up domain and create content
```
I believe we use two words when used as a verb, one word when used as a noun

### Secure the domain

> First step is to request @beyang or @sqs get a domain and add the site to Sourcegraph's Cloudflare account. 

Reviewer: ```suggestion
First step is to request @beyang get a domain and add the site to Sourcegraph's Cloudflare account. 
```

    
    💡 Make sure `www` is redirecting to `https`.

### Create the website content

Once the domain is secured, we need to setup the GitHub page, where the static page will be hosted as: `[projectname].github.io`.

> We clone the repo locally, create a branch from master (make sure master is up-to-date!) and add the following files:

Reviewer: ```suggestion
We clone the repository locally, create a branch from master (make sure master is up-to-date!) and add the following files:
```

    - CNAME
    Containing:

        [project url]

    - README.md
    - index.html
    - style-addition.css

When all the content is created, we `push` our branch, create a **PR** and ask for reviews and merge to master.

> Examples: [Langserver](http://github.com/langserver), [LSIF](http://github.com/lsif)

Reviewer: These appear to be links to the orgs, not the repos

# 3. Update the DNS to point to the server

> We use terraform to manage our DNS records. To update DNS, follow the [DNS guide](../../../../../../infrastructure/blob/master/dns/README.md), but push to `branch` not `master` and get it reviewed before merging.

Reviewer: ```suggestion
We use terraform to manage our DNS records. To update DNS, follow the [DNS guide](../../../../../../infrastructure/blob/master/dns/README.md), but push to a non-`master` branch and get it reviewed before merging.
```
Ignore this if I'm misinterpreting this one

> We use terraform to manage our DNS records. To update DNS, follow the [DNS guide](../../../../../../infrastructure/blob/master/dns/README.md), but push to `branch` not `master` and get it reviewed before merging.

Author: The style guide for documentation states to use relative paths. Should I make an exemption here since it's so far out?

> We use terraform to manage our DNS records. To update DNS, follow the [DNS guide](../../../../../../infrastructure/blob/master/dns/README.md), but push to `branch` not `master` and get it reviewed before merging.

Reviewer 2: Relative paths only make sense within the sourcegraph/sourcegraph repo. This link will break if I try to click on this link locally and don't have sourcegraph/infrastructure checked out as a sibling. Can you update the docs that you are referring to to clarify this?

If you don't have terraform installed and install via ```brew install terraform```:

>     💡 Ensure that your local `terrraform` version is the same as the `CI` version to **avoid file locks**.

Reviewer: ```suggestion
    💡 Ensure that your local `terraform` version is the same as the `CI` version to **avoid file locks**.
```

    💡 Make sure to pull the **latest version** of the infrastructure `master` before creating your branch.

>     💡 Check that all of the domain's `A records` are also referenced in the terraform file.

Reviewer 2: The formatting here seems a bit unusual. Can this be an ordered/unordered list?

> Inside your branch:

Reviewer: ```suggestion
From the `infrastructure` repository, with your branch checked out:
```

    $ terraform apply
    $ git commit -m 'dns: Updated terraform.tfstate' terraform.tfstate
    $ git push origin branch

Get the branch reviewed and merge to `master`.

> Other Sourcegraph docs:

Reviewer: Hah — this is new. Is the idea just to list random docs, or are these somehow related to this particular one?

Author: Yes, sorry, I thought that this was the reason why the build keeps failing as I get this error:
```
dev/documentation/separate_website.md: disconnected page (no inlinks from other pages)
--
  | [2019-10-15T01:07:37Z] 1 problems found

```
So I thought maybe I need to mention other doc links for it to pass, but I guess that doesn't solve it  😅. I can remove it again. Any idea what the build error means?

Reviewer 2: You need to link *to* this page from somewhere (e.g. from index.md)
